pkg/extprom/http/instrument_server.go

```diff
@@ -0,0 +1,107 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// ServerInstrumentor holds necessary metrics to instrument an http.Server
+// and provides necessary behaviors
+type ServerInstrumentor interface {
```

Comment:
Also not sure of `Instrumentor` name.. Isn't `Middleware` bit better and clearer? Not a blocker, might be by personal preference, up to you.

Reply:
Maybe I am wrong but I think that if we would name these middlewares then there might be more confusion in the future when (if) we will switch to using Cortex's middlewares for caching.

Reply:
I wasn't also sure about `Instrumentor` name, it's not even an actual word. I'm ok to use `Middleware`.

@GiedriusS Since it will be under a particular namespace with package structure, maybe it won't create any confusion. What do you think?

Reply:
@bwplotka How about `InstrumentationMiddleware`? It'll go like `extpromhttp.NewInstrumentationMiddleware`, `NewInstrumentationMiddleware().NewHandler()` etc.

Reply:
Again, don't have strong opinion (: Happy with `Instrumentor`

Reply:
> Maybe I am wrong but I think that if we would name these middlewares then there might be more confusion in the future when (if) we will switch to using Cortex's middlewares for caching.

Well that will be exactly the same middleware isn't it?

```
 ... handler http.Handler) http.HandlerFunc { ...
```